File: common/src/main/java/com/hartwig/actin/molecular/datamodel/driver/GeneAlteration.java
package com.hartwig.actin.molecular.datamodel.driver;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface GeneAlteration {

    @NotNull
    String gene();

    @NotNull
    GeneRole geneRole();

    @NotNull
    ProteinEffect proteinEffect();

    @Nullable
    Boolean associatedWithDrugResistance();
}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/driver/GeneAlteration.kt
package com.hartwig.actin.datamodel.molecular.driver

import com.hartwig.actin.datamodel.molecular.driver.ProteinEffect

interface GeneAlteration {
    val gene: String
    val geneRole: GeneRole
    val proteinEffect: ProteinEffect
    val isAssociatedWithDrugResistance: Boolean?
}
